plugins/Unity/Launcher/appdrawermodel.cpp

```diff
@@ -82,11 +84,12 @@ void AppDrawerModel::appAdded(const QString &appId)
     endInsertRows();
 }
 
-void AppDrawerModel::appRemoved(const QString &appId)
+// ubuntuAppId is appID without version suffix, see xdgwatcher.cpp.
+void AppDrawerModel::appRemoved(const QString &appId, const QString &ubuntuAppId)
```

Comment:
Instead of passing both in here, why not make the functions (which seem to not require the class object itself to be useful), into public static methods, and then just call `XdgWatcher::stripAppIdVersion(appid)` when needed below? This avoids a semi-duplicated argument, and makes the code easier to follow.

Reply:
problem with that is that the "normal/standard" appID might not be the same as strippedAppId as the "ubuntu style" app id does not add the parent folder into app id.

example:
.local/share/desktop/kde/kate.desktop would be kde-kate as appID.

Reply:
example where these conflics:

.local/share/desktop/ubports/system-settings_ubports_1.1.1.desktop would then be passed as ubports-system-settings_ubports_1.1.1 to appRemoved, then stripped to ubports-system-settings_ubports and that would be wrong, as "ubuntu style" appID does not append "folder name" as xdg one does

Reply:
But, click hook does not create desktop files in a subdir, and we get the app ID for these from inside the desktop file, so this isn't an issue. I don't see any need to have extra special handling in here. The model has a list of app IDs, and one gets removed. there doesn't need to be extra logic for special cases in here.